client/classic/upe/index.js

```diff
@@ -0,0 +1,634 @@
+		).first();
+
+		if ( ! $container.length ) {
+			return;
+		}
+
+		// Adapted from WooCommerce core @ ea9aa8c, assets/js/frontend/checkout.js#L514-L529
+		$(
+			'.woocommerce-NoticeGroup-checkout, .woocommerce-error, .woocommerce-message'
+		).remove();
+		$container.prepend(
+			'<div class="woocommerce-NoticeGroup woocommerce-NoticeGroup-checkout">' +
+				messageWrapper +
+				'</div>'
+		);
+		$container
+			.find( '.input-text, select, input:checkbox' )
+			.trigger( 'validate' )
+			.blur();
+
+		let scrollElement = $( '.woocommerce-NoticeGroup-checkout' );
+		if ( ! scrollElement.length ) {
+			scrollElement = $container;
+		}
+
+		$.scroll_to_notices( scrollElement );
+		$( document.body ).trigger( 'checkout_error' );
+	};
+
+	// Show or hide save payment information checkbox
+	const showNewPaymentMethodCheckbox = ( show = true ) => {
+		if ( show ) {
+			$( '.woocommerce-SavedPaymentMethods-saveNew' ).show();
+		} else {
+			$( '.woocommerce-SavedPaymentMethods-saveNew' ).hide();
+			$( 'input#wc-woocommerce_payments-new-payment-method' ).prop(
+				'checked',
+				false
+			);
+		}
+	};
+
+	// Set the selected UPE payment type field
+	const setSelectedUPEPaymentType = ( paymentType ) => {
+		$( '#wc_stripe_selected_upe_payment_type' ).val( paymentType );
+	};
+
+	/**
+	 * Converts form fields object into Stripe `billing_details` object.
+	 *
+	 * @param {Object} fields Object mapping checkout billing fields to values.
+	 * @return {Object} Stripe formatted `billing_details` object.
+	 */
+	const getBillingDetails = ( fields ) => {
+		return {
+			name: `${ fields.billing_first_name } ${ fields.billing_last_name }`.trim(),
+			email: fields.billing_email,
+			phone: fields.billing_phone,
+			address: {
+				country: fields.billing_country,
+				line1: fields.billing_address_1,
+				line2: fields.billing_address_2,
+				city: fields.billing_city,
+				state: fields.billing_state,
+				postal_code: fields.billing_postcode,
+			},
+		};
+	};
+
+	/**
+	 * Mounts Stripe UPE element if feature is enabled.
+	 *
+	 * @param {boolean} isSetupIntent {Boolean} isSetupIntent Set to true if we are on My Account adding a payment method.
+	 */
+	const mountUPEElement = function ( isSetupIntent = false ) {
+		// Do not mount UPE twice.
+		if ( upeElement || paymentIntentId ) {
+			return;
+		}
+
+		// If paying from order, we need to create Payment Intent from order not cart.
+		const isOrderPay = getStripeServerData()?.isOrderPay;
+		const isCheckout = getStripeServerData()?.isCheckout;
+		let orderId;
+		if ( isOrderPay ) {
+			orderId = getStripeServerData()?.orderId;
+		}
+
+		const intentAction = isSetupIntent
+			? api.initSetupIntent()
+			: api.createIntent( orderId );
+
+		const $upeContainer = $( '#wc-stripe-upe-element' );
+		blockUI( $upeContainer );
+
+		intentAction
+			.then( ( response ) => {
+				// I repeat, do NOT mount UPE twice.
+				if ( upeElement || paymentIntentId ) {
+					unblockUI( $upeContainer );
+					return;
+				}
```

Comment:
I find it sort of weird (not too weird though tbh) that we have to check for this twice.

Is this indicative of some other issue, or is it truly "normal" to have a situation where you might be mounting the UPE element twice? It seems to indicate that we might be sending the same request twice at some point? Maybe that's something we should be handling instead of checking whether the element is mounted twice?

Reply:
This comes directly from the WCPay implementation. I don't know the answer to your questions, but I was wondering the same thing when I saw it there.

Reply:
Just so we're not holding up this PR, could you create an issue to look into this further? Feel free to just link to this discussion with a short description of what we want to look into. I think the priority can be set to `low` 🙂

Reply:
[Done.](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1689)